### PR TITLE
Add Supermaven support

### PR DESCRIPTION
**Pull request by lino-levan**
Release Notes:
- Closes ([#10013](https://github.com/zed-industries/zed/issues/10013)).

Current UI (will probably be removed, but just mocking up this feature before I get it functionally working).

image

### CODE REVIEW

File: assets/icons/zed_x_supermaven.svg
Comment (maxdeviant) [inline assets/icons/zed_x_supermaven.svg:1]: Looks like Supermaven has a new logo now that has less... problematic... similarities with a certain other logo:

Screenshot 2024-04-01 at 10 59 58 AM

Comment (lino-levan) [reply assets/icons/zed_x_supermaven.svg:1]: Yup, just saw that. I'm gonna go ahead and redo the art.

**Comment by nathansobo** [comment]
Hey! Thanks so much for opening this. We're meeting with Supermaven this week and getting clarity on what a partnership might look like. That will give us more information as to how we want to proceed from a UI perspective. We'll touch base later in the week.

**Comment by lino-levan** [comment]
Sounds good.

**Comment by nathansobo** [comment]
I talked to Jacob, and we've decided to ship this as the default experience, with the option of doing copilot if you want. They're adding an API for us to interface with on the backend to use of the free version for anybody who authenticates in Zed. They need us to collect an email address though, so that's going to take some adjustment. It will take a bit of time to integrate. Maybe you can help, but it's also a bit in the weeds of our internal details here.

**Comment by lino-levan** [comment]
I just wanted to help push the effort forward. If the team wants to ship this in the default experience, that would be great. Closing this PR for now since it doesn't really ship any value, but please let me know if I can help!

This + ghostty integration is the headline features I'm most excited about.

**Pull request closed without merging**